crypto/crypto.go

```diff
@@ -122,12 +124,13 @@ func FromECDSA(priv *ecdsa.PrivateKey) []byte {
 	return math.PaddedBigBytes(priv.D, priv.Params().BitSize/8)
 }
 
-func ToECDSAPub(pub []byte) *ecdsa.PublicKey {
-	if len(pub) == 0 {
-		return nil
-	}
+// UnmarshalPubkey converts bytes to a secp256k1 public key.
+func UnmarshalPubkey(pub []byte) (*ecdsa.PublicKey, error) {
 	x, y := elliptic.Unmarshal(S256(), pub)
-	return &ecdsa.PublicKey{Curve: S256(), X: x, Y: y}
+	if x == nil {
```

Comment:
I would also check that `y != nil`, because if it's true that in the current implementation `x` and `y` are either both `nil` or both non-`nil`, this is an undocumented feature and it could change in the future (btw, the fact that `elliptic.Unmarshal` returns `x == nil` on error is a bit strange to me, in the context of Go, which is what prompts me to write this nitpick because the interface should change)

Reply:
package elliptic is a standard library package and returning nil on failure is a known shortcoming of that API. That's one of the reasons we have the wrapper in the first place.